src/glum/_glm.py

```diff
@@ -226,6 +228,53 @@ def _name_categorical_variables(
     return new_names
 
 
+def _parse_formula(
+    formula: Union[str, Formula], include_intercept: bool = True
```

Comment:
You could use `from formulaic import FormulaSpec` and `def _parse_formula(formula: FormulaSpec, ...)` here for greater generality :).

Reply:
Very cool, thanks for the tip!